Cython/Compiler/Visitor.py

```diff
@@ -659,6 +660,7 @@ def _dispatch_to_handler(self, node, function, arg_list, kwargs):
     def _dispatch_to_method_handler(self, attr_name, self_arg,
                                     is_unbound_method, type_name,
                                     node, function, arg_list, kwargs):
+        type_name = punycodify_name(type_name) # FIXME: this seems like it should be handled much earlier
```

Comment:
Why is this needed?

Reply:
I think I'd misidentified the problem slightly. _get_handler does `getattr` with a unicode string. This isn't valid when running Cython in Python 2 since only ascii strings are allowed. I've adopted a different approach of returning None from _get_handler if it's passed a unicode string. I'm not sure I understand the purpose of handlers enough to know if this is always valid, but it shouldn't break any existing code (since existing code should be passing unicode strings)